flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/source/ContinuousFileReaderOperatorTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.functions.source;
+
+import org.apache.flink.api.common.io.FileInputFormat;
+import org.apache.flink.core.fs.FileInputSplit;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.streaming.runtime.tasks.StreamTaskActionExecutor;
+import org.apache.flink.streaming.runtime.tasks.TestProcessingTimeService;
+import org.apache.flink.streaming.runtime.tasks.mailbox.Mail;
+import org.apache.flink.streaming.runtime.tasks.mailbox.MailboxExecutorImpl;
+import org.apache.flink.streaming.runtime.tasks.mailbox.TaskMailboxImpl;
+
+import org.junit.Test;
+
+/**
+ * {@link ContinuousFileReaderOperator} test.
+ */
+public class ContinuousFileReaderOperatorTest {
```

Comment:
Why have you created a separate class to add this unit test? There is already a similar test `org.apache.flink.hdfstests.ContinuousFileProcessingTest#testExceptionHandling` so definitely those two should be together. 

A good questions is why is the test for a class `ContinuousFileReaderOperator` named `ContinuousFileProcessingTest` but I guess that's a story for another time. (I would be +1 for renaming all `ContinuousFileProcessing***` classes to `ContinuousFileReaderOperator***`)

Reply:
I added a unit test while `ContinuousFileProcessingTest` is an integration test. And it also resides in a different project (because of Hadoop dependencies).

Reply:
Hmmm, ok I see. I missed that it's in a different package. I think I would still slightly prefer to keep the tests together in the `ContinuousFileProcessingTest` class, and in the long run someone should reimplement those tests without relaying on Hadoop.

But if you would like to keep the unit tests closer to the actual class without doing/waiting for the refactor, in that case please also move `ContinuousFileProcessingTest#testExceptionHandling` to the new `ContinuousFileReaderOperatorTest` class.